### PR TITLE
support REPL modes 

### PR DESCRIPTION
**Pull request by MHellmund**
support  REPL help-, shell-, and pkg-mode, 
similar to IJulia for one-line input cells only,  
see https://github.com/PumasAI/QuartoNotebookRunner.jl/issues/56

Examples:

    ```{julia}
    ;ls
    ```

    ```{julia}
    ?for
    ```

    ```{julia}
    ]status
    ```

### CODE REVIEW

File: src/server.jl [hunk 1]
```diff
@@ -70,6 +70,8 @@ function _exeflags_and_env(options)
     if !any(startswith("JULIA_PROJECT="), env) && !any(startswith("--project="), exeflags)
         push!(exeflags, "--project=@.")
     end
+    # if exeflags already contains '--color=no', the 'no' will prevail
+    pushfirst!(exeflags,"--color=yes")
```
Comment (MichaelHatherly) [inline src/server.jl:74]: Related to #69 I assume? Can this be moved to a separate PR, which we'll be able to review and merge separately.

**Review by MichaelHatherly — CHANGES_REQUESTED** [review]

File: src/server.jl (hunk 1 above)
Comment (MHellmund) [reply src/server.jl]: oops, I removed the second commit

**Comment by MichaelHatherly** [comment]
Thanks @MHellmund. This'll need some tests prior to merging. Let me know if you're unsure at all about the test folder structure.

**Comment by MHellmund** [comment]
I added a test. I had to add `REPL` to `test/Project.toml`. Without it, the test produces an error: [repl_modes.docx](https://github.com/PumasAI/QuartoNotebookRunner.jl/files/15226633/repl_modes.docx)   ?

File: src/worker.jl [hunk 2]
```diff
@@ -396,6 +396,34 @@ function worker_init(f::File)
...
+            # "] ..." cells are interpreted as pkg shell commands
+            hcode = replace(code, r"^\]" => "")
+            if hcode != code
+                code = string(
+                    "Pkg.REPLMode.PRINTED_REPL_WARNING[]=true; Pkg.REPLMode.do_cmd(Pkg.REPLMode.MiniREPL(),\"",
+                    chomp(hcode),
+                    "\")",
+                )
+            end
+
```
Comment (MichaelHatherly) [inline src/worker.jl:426]: Only one of these is valid at any one time, right? If that's the case then the `replace` shouldn't be run for all three every single cell, just the one that matches.

File: test/Project.toml [hunk 3]
```diff
@@ -3,6 +3,7 @@ JSON3 = "0f8b85d8-7281-11e9-16c2-39a750bddbf1"
 JSONSchema = "7d188eb4-7ad8-530c-ae41-71a32a6d4692"
 Logging = "56ddb016-857b-54e1-b83d-db4d58db5568"
 NodeJS_18_jll = "c1e1d063-8311-5f52-a749-c7b05e91ae37"
+REPL = "3fa0cd96-eef1-5676-8a61-b3b8758bbffb"
```
Comment (MichaelHatherly) [inline test/Project.toml:6]: Having to do this isn't ideal. Were there no alternatives available to avoid it?

**Review by MichaelHatherly — CHANGES_REQUESTED** [review]
Thanks @MHellmund, left some comments inline.

The code building via strings isn't ideal, might be more robust to do it via expressions instead. That would likely also allow avoiding the `import REPL` issue.

File: test/Project.toml (hunk 3 above)
Comment (MichaelHatherly) [reply test/Project.toml:6]: So `REPL` is always loaded, since we use it in: https://github.com/PumasAI/QuartoNotebookRunner.jl/blob/a003fa8bbec452dfc79ba85c9fd279643198ee76/src/worker.jl#L12

That is the `Main` module where it's loaded. So the very straight-forward fix is to do `Main.REPL` wherever you have `REPL` currently rather than `import`ing it into the cell. See if that helps avoid adding the dep explicitly.

**Comment by MHellmund** [comment]
New version still operates on with the `code`  string, but more efficient;  problem with missing REPL module solved (thanks!).
The old version allowed whitespace in front of '?' and ';' . Since the REPL itself doesn't, I changed that.

**Review by MichaelHatherly — CHANGES_REQUESTED** [review]

File: src/worker.jl [hunk 4]
```diff
@@ -396,6 +396,17 @@ function worker_init(f::File)
...
             loc = LineNumberNode(line, Symbol(file))
+
+            # handle REPL modes
+            if code[1] == '?'
+                code = "Core.eval(Main.REPL, Main.REPL.helpmode(\"$(code[2:end])\"))"
+            elseif code[1] == ';'
+                code = "Base.repl_cmd(`$(code[2:end])`, stdout)"
+            elseif code[1] == ']'
+                code = "Pkg.REPLMode.PRINTED_REPL_WARNING[]=true;\
+                        Pkg.REPLMode.do_cmd(Pkg.REPLMode.MiniREPL(),\"$(code[2:end])\")"
```
[the remark this answers is not in the document]
Comment (MHellmund) [reply src/worker.jl:407]: ah, this was introduced in julia 1.7 as a method to break up long string literals...

**Review by MichaelHatherly — APPROVED** [review]
Thanks @MHellmund.
